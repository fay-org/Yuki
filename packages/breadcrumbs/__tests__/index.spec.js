import { mount } from '@vue/test-utils'
import Breadcrumbs from '../index'
import BreadcrumbsItem from '../../breadcrumbs-item'

describe('Breadcrumbs', () => {
  it('should be render as element', () => {
    const braedcrumbs = mount(<Breadcrumbs />)
    expect(() => braedcrumbs.unmount()).not.toThrow()
  })

  it('should be support different sizes', () => {
    const wrapper = mount({
      setup() {
        return { sizes: ['mini', 'small', 'medium', 'large'] }
      },
      render() {
        return (
          <>
            {this.sizes.map((size, idx) => (
              <Breadcrumbs size={size} key={idx}>
                {idx}
              </Breadcrumbs>
            ))}
          </>
        )
      },
    })

    const els = wrapper.findAll('.fect-breadcrumbs')
    expect(els[0].attributes('style')).toBe('font-size: 12px;')
    expect(els[1].attributes('style')).toBe('font-size: 14px;')
    expect(els[2].attributes('style')).toBe('font-size: 16px;')
    expect(els[3].attributes('style')).toBe('font-size: 18px;')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be support custom separator', () => {
    const wrapper = mount(
      <Breadcrumbs separator=">">
        <BreadcrumbsItem />
        <BreadcrumbsItem />
      </Breadcrumbs>,
    )
    expect(wrapper.find('.fect-breadcrumbs__separator').text()).toBe('>')
  })
})
