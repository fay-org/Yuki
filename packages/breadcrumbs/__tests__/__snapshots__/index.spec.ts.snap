// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should be support different sizes 1`] = `
<nav class="fect-breadcrumbs" style="font-size: 12px;"> {idx} </nav>
<nav class="fect-breadcrumbs" style="font-size: 14px;"> {idx} </nav>
<nav class="fect-breadcrumbs" style="font-size: 16px;"> {idx} </nav>
<nav class="fect-breadcrumbs" style="font-size: 18px;"> {idx} </nav>
`;
